config-store-api/src/main/java/io/stargate/config/store/api/ConfigWithOverrides.java

```diff
@@ -45,20 +47,35 @@ public ConfigWithOverrides(@Nonnull Map<String, Object> configMap) {
    *
    * <p>3. Underlying config map
    *
+   * <p>When trying to retrieve the system property and environment variable, it will use the full
+   * setting name. It will add the module name for which this `ConfigWithOverrides` is created.
+   *
+   * <p>For example, if the {@code String moduleName = "m_1"} and you are calling this
+   * getWithOverrides() method for settingName = "s" it will firstly try to {@code
+   * System.getProperty("m_1.s")}, then {@code System.getenv("m_1.s")} and finally get the config
+   * from the underlying map using {@code Map.get("s")}.
+   *
+   * <p>Prefixing with module name is done to avoid conflicts of overrides between modules.
+   *
    * @return the value with the highest priority or null if there is no value associated with the
    *     given settingName.
    */
   @Nullable
   public Object getWithOverrides(String settingName) {
```

Comment:
Can we have two versions of this? One that adds the module prefix and another that uses the exact settingName as provided?

Reply:
Some of these are also not module specific like `stargate.auth_id` and `stargate.persistence_id`

Reply:
I think that we should teat those setting as settings per the stargate module, so the client of this API would do:
`ConfigStore.getConfigForModule("stargate"). getWithOverrides("auth_id")`
by using this mechanism we can provide a map of setting global to stargate in the `stargate-config.yaml`.
So I don't see a need for the 2nd version of this method.